ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                        )
+                    });
+                }
+
+                yield return instance;
+            }
+        }
+
+        /// <summary>
+        /// Writes the processed data into their respective files
+        /// </summary>
+        /// <param name="news">Enumerable containing news articles</param>
+        /// <param name="date">Date to convert for</param>
+        private void WriteToFile(IEnumerable<BenzingaNews> news, DateTime date)
+        {
+            var finalDirectory = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, "content", $"{date:yyyMMdd}"));
+            finalDirectory.Create();
+
+            foreach (var article in news)
+            {
+                var createdReference = false;
+
+                // For each ticker, add a reference to the article we're going to store under the "content" folder
+                foreach (var ticker in article.Symbols)
+                {
+                    // Invalid character in path
+                    if (ticker.Symbol.Value.Contains(":"))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} contains invalid character ':'. Skipping");
+                        continue;
+                    }
+                    // ETFs have a special exchange tag associated with them.
+                    // Sometimes, we have an empty exchange for some tickers. Let's try and map them first before giving up on them.
+                    // Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd
+                    if (ticker.Exchange != "NASDAQ" && ticker.Exchange != "NYSE" && ticker.Exchange != "ETF" && !string.IsNullOrEmpty(ticker.Exchange))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is not in NYSE, NASDAQ, or is not an ETF. Skipping");
+                        continue;
+                    }
+                    // We can't write these tickers in Windows, so we'll skip them
+                    if (_isWindows && _forbiddenTickers.Contains(ticker.Symbol.Value))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is invalid in Windows. Skipping");
+                        continue;
+                    }
+
+                    var tickerPath = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, ticker.Symbol.Value.ToLowerInvariant()));
+                    tickerPath.Create();
+
+                    var referenceFilePath = Path.Combine(tickerPath.FullName, $"{date:yyyMMdd}.csv");
+                    var tickerReferences = new HashSet<string>();
+
+                    if (File.Exists(referenceFilePath))
+                    {
+                        tickerReferences = File.ReadAllLines(referenceFilePath)
+                            .ToHashSet();
```

Comment:
style nit -- I would throw the `ToHashSet()` on the same line

not a nit -- **why** are we using a `HashSet<T>` here at all? hash sets, or really sets in general, are great for two reasons, one is guaranteeing uniqueness and for `O(1)`  `Contains`/`Exists` calls. Do we expect the file at `referenceFilePath` to contain duplicates?

I suppose even if we _don't_ expect dupes, I'm sure it's nice being able to guarantee that no dupes will be written. it's nice to define _invariants_ like that -- helps you reason about code and more importantly, and downstream applications that might pick up this application's output -- such as algorithms 😄